node/src/storage_mngr.rs

```diff
@@ -35,8 +41,13 @@ pub fn start_from_config(config: Config) {
 pub fn get<K, T>(key: &K) -> impl Future<Output = Result<Option<T>, failure::Error>>
 where
     K: serde::Serialize,
-    T: serde::de::DeserializeOwned,
+    T: serde::de::DeserializeOwned + 'static,
 {
+    // Check that we don't accidentally use this function with some certain special types
+    if TypeId::of::<T>() == TypeId::of::<ChainState>() {
+        panic!("Please use get_chain_state instead");
```

Comment:
I added this panic to ensure that the correct method is used. I guess this could be turned into a compile-time error, but no idea how.

Reply:
me neither